### PR TITLE
refactor(app): create generic task list component with progress tracker

### PR DESCRIPTION
**Pull request by brenthagen**
# Overview

adds a generic task list component with associated progress tracker, intended to be used in the calibration dashboard. left todos if in the future we want to refactor the layout of the task connector or internals of the subcomponent props.

closes RAUT-196, closes RAUT-101

# Changelog

  - Creates generic task list component

# Review requests

do the active index and task interfaces make sense? 

# Risk assessment

low

### CODE REVIEW

File: app/src/organisms/TaskList/index.tsx [hunk 1]
```diff
@@ -0,0 +1,408 @@
...
+  const isPastTask = activeTaskIndex != null && taskIndex < activeTaskIndex
+  const isLastTask = taskIndex === taskListLength - 1
+  const hasSubTasks = subTasks.length > 0
+  const isActiveTaskWithSubtasks = taskIndex === activeTaskIndex && hasSubTasks
+  const isFutureTask = activeTaskIndex != null && taskIndex > activeTaskIndex
+
+  // a connector between task icons
+  const taskConnector = (
+    <Flex
+      flex={1}
```
Comment (koji) [inline app/src/organisms/TaskList/index.tsx]:
```suggestion
      flex="1"
```

File: app/src/organisms/TaskList/index.tsx [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+
+  return (
+    <Flex flexDirection={DIRECTION_COLUMN} alignItems={ALIGN_CENTER}>
+      {isComplete || isTaskListComplete || isPastTask ? (
+        <Icon
+          size={SPACING.spacing4}
+          margin={SPACING.spacing4}
+          name="ot-check"
+          height="20px"
+          width="20px"
```
Comment (koji) [inline app/src/organisms/TaskList/index.tsx:75]: maybe
```suggestion
size="1.25rem"
```

File: app/src/organisms/TaskList/index.tsx [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+    />
+  )
+
+  const noSubTaskConnector = !isLastTask ? taskConnector : null
+
+  return (
+    <Flex flexDirection={DIRECTION_COLUMN} alignItems={ALIGN_CENTER}>
+      {isComplete || isTaskListComplete || isPastTask ? (
+        <Icon
+          size={SPACING.spacing4}
```
Comment (koji) [inline app/src/organisms/TaskList/index.tsx]: Do we still need `width` and `height` when we use `size`?

File: app/src/organisms/TaskList/index.tsx [hunk 4]
```diff
@@ -0,0 +1,408 @@
...
+          flex={FLEX_NONE}
+          alignItems={ALIGN_CENTER}
+          justifyContent={JUSTIFY_CENTER}
+          backgroundColor={
+            isFutureTask ? COLORS.medGreyHover : COLORS.blueEnabled
+          }
+          color={COLORS.white}
+          margin={SPACING.spacing4}
+          height="20px"
+          width="20px"
```
Comment (koji) [inline app/src/organisms/TaskList/index.tsx]: maybe
```suggestion
          height="1.25rem"
          width="1.25rem"
```

File: app/src/organisms/TaskList/index.tsx [hunk 5]
```diff
@@ -0,0 +1,408 @@
...
+                  borderWidth={SPACING.spacing1}
+                  color={COLORS.white}
+                  margin={SPACING.spacing4}
+                  height="12px"
+                  width="12px"
+                  borderRadius="6px"
+                />
+                {/* subtask connector component */}
+                <Flex
+                  flex={1}
```
Comment (koji) [inline app/src/organisms/TaskList/index.tsx]:
```suggestion
                  flex="1"
```

File: app/src/organisms/TaskList/index.tsx [hunk 6]
```diff
@@ -0,0 +1,408 @@
...
+                  }
+                  border={BORDERS.lineBorder}
+                  borderColor={
+                    isFutureSubTask ? COLORS.medGreyHover : COLORS.blueEnabled
+                  }
+                  borderWidth={SPACING.spacing1}
+                  color={COLORS.white}
+                  margin={SPACING.spacing4}
+                  height="12px"
+                  width="12px"
```
Comment (koji) [inline app/src/organisms/TaskList/index.tsx]: maybe
```suggestion
                  height="0.75rem"
                  width="0.75rem"
```

**Review by koji — APPROVED** [review]
Ran this on Storybook. Looks awesome!
Thank you for doing this. (This might be the most complicated UI component in this repo 😁)

File: app/src/organisms/TaskList/index.tsx (hunk 3 above)
Comment (brenthagen) [reply app/src/organisms/TaskList/index.tsx]: i don't think so, thanks

**Review by b-cooper — APPROVED** [review]
This looks great!  super clear commenting and code too.  I can't wait to use it!

**Pull request merged**